portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +36,57 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  /** Returns all questions posted */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Jason</h1>");
+    Query query = new Query("Questions").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Question> questions = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty("name");
+      String text = (String) entity.getProperty("text");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Question userQuestion = new Question(name, text, timestamp);
+      questions.add(userQuestion);
+    }
+
+    response.setContentType("application/json;");
+    response.getWriter().println(convertToJSON(questions));
+  }
+
+  /** Posts the users questions on a message board. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String name = request.getParameter("name");
+    String text = request.getParameter("text");
+    long timestamp = System.currentTimeMillis();
+
+    Entity questionEntity = new Entity("Questions");
+    questionEntity.setProperty("name", name);
```

Comment:
the property names "name" "text" "timestamp" can probably be moved to private static final String constants